frontend/src/components/Secrets/GSecretDialogDDNS.vue

```diff
@@ -0,0 +1,245 @@
+        </p>
+      </div>
+    </template>
+  </g-secret-dialog>
+</template>
+
+<script>
+import { useVuelidate } from '@vuelidate/core'
+import { required } from '@vuelidate/validators'
+
+import GSecretDialog from '@/components/Secrets/GSecretDialog'
+import GExternalLink from '@/components/GExternalLink'
+
+import { getErrorMessages } from '@/utils'
+import {
+  withFieldName,
+  withMessage,
+} from '@/utils/validators'
+
+export default {
+  components: {
+    GSecretDialog,
+    GExternalLink,
+  },
+  props: {
+    modelValue: {
+      type: Boolean,
+      required: true,
+    },
+    secret: {
+      type: Object,
+    },
+  },
+  emits: [
+    'update:modelValue',
+  ],
+  setup () {
+    return {
+      v$: useVuelidate(),
+    }
+  },
+  data () {
+    return {
+      server: undefined,
+      tsigKeyName: undefined,
+      tsigSecret: undefined,
+      zone: undefined,
+      tsigSecretAlgorithm: 'hmac-sha256',
+      hideTSIGSecret: true,
+      secretAlgorithmItems: [
+        {
+          title: 'HMAC-SHA256 (default)',
+          value: 'hmac-sha256',
+        },
+        {
+          title: 'HMAC-SHA224',
+          value: 'hmac-sha224',
+        },
+        {
+          title: 'HMAC-SHA384',
+          value: 'hmac-sha384',
+        },
+        {
+          title: 'HMAC-SHA512',
+          value: 'hmac-sha512',
+        },
+        {
+          title: 'HMAC-SHA1',
+          value: 'hmac-sha1',
+        },
+        {
+          title: 'HMAC-MD5',
+          value: 'hmac-md5',
+        },
+      ],
+    }
+  },
+  validations () {
+    return {
+      server: withFieldName('Server', {
+        required,
+        format: withMessage('Must have format <host>:<port>', value => {
+          return /^([^:]+):(\d+)$/.test(value)
+        }),
+      }),
+      tsigKeyName: withFieldName('TSIG Key Name', {
+        required,
+        format: withMessage('Must end with a dot', value => {
+          return /\.$/.test(value)
+        }),
+      }),
+      tsigSecret: withFieldName('TSIG Secret', {
+        required,
+      }),
+      zone: withFieldName('Zone', {
+        required,
+        format: withMessage('Must be fully qualified', value => {
+          return /^[a-zA-Z0-9-]+(\.[a-zA-Z0-9-]+)*\.$/.test(value)
+        }),
+      }),
+    }
+  },
+  computed: {
+    visible: {
+      get () {
+        return this.modelValue
+      },
+      set (modelValue) {
+        this.$emit('update:modelValue', modelValue)
+      },
+    },
+    valid () {
+      return !this.v$.$invalid
+    },
+    secretData () {
+      const data = {
+        Server: this.server,
+        TSIGKeyName: this.tsigKeyName,
+        TSIGSecret: this.tsigSecret,
+        Zone: this.zone,
+      }
+      if (this.tsigSecretAlgorithm !== 'hmac-sha256') {
+        data.TSIGSecretAlgorithm = this.tsigSecretAlgorithm
+      }
+      return data
+    },
+    isCreateMode () {
+      return !this.secret
+    },
+  },
+  watch: {
+    value: function (value) {
```

Comment:
Leftover from vue3 migration. It is not causing an issue because the dialog is always re-mounted when re-opened because of the way it is created using `<component :is>`. Added a comment so that we not accidentally change the behavior and removed all watches and reset logic.